src/core/ack_tracker.c

```diff
@@ -96,6 +96,64 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+//
+// This function implements the logic defined in Section 6.2 [draft-ietf-quic-frequence-10]
+// to determine if the reordering threshold has been hit.
+//
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
+    _In_ QUIC_ACK_TRACKER* Tracker,
+    _In_ uint8_t ReorderingThreshold
+    )
+{
+    if (ReorderingThreshold == 0) {
+        return FALSE;
+    }
+
+    const uint64_t LargestUnacked = QuicRangeGetMax(&Tracker->PacketNumbersToAck);
+    uint64_t LargestReported;  // The largest packet number that could be declared lost 
+
+    //
+    // Largest Reported is equal to the largest packet number acknowledged minus the Reordering Threshold.
+    // If the difference between the largest packet number acknowledged and the Reordering Threshold is smaller than the
+    // smallest packet in the ack tracker, then the largest reported is the smallest packet in the ack tracker.
+    //
+
+    if (Tracker->LargestPacketNumberAcknowledged >= QuicRangeGet(&Tracker->PacketNumbersToAck, 0)->Low + ReorderingThreshold) {
+        LargestReported = Tracker->LargestPacketNumberAcknowledged - ReorderingThreshold + 1;
+    } else {
+        LargestReported = QuicRangeGet(&Tracker->PacketNumbersToAck, 0)->Low;
+    }
+
+    //
+    // Loop through all previous ACK ranges (before last) to find the smallest missing
+    // packet number that is after the largest reported packet number. If the difference
+    // between that missing number and the largest unack'ed number is more than the
+    // reordering threshold, then the condition has been met to send an immediate
+    // acknowledgement.
+    //
+
+    for (uint32_t Index = QuicRangeSize(&Tracker->PacketNumbersToAck) - 1; Index > 0; --Index) {
+        uint64_t SmallestMissing = QuicRangeGetHigh(QuicRangeGet(&Tracker->PacketNumbersToAck, Index - 1)) + 1; // Smallest missing in the previous gap
+        uint64_t RangeStart = QuicRangeGet(&Tracker->PacketNumbersToAck, Index)->Low; // Lowest Packet number in the subrange
```

Comment:
My personal preference is to swap the order of the definitions of each of these variables. Also, I think the comments on each are probably unnecessary. Finally, let's make them `const`. While mostly unnecessary, it's nice to declare variables `const` if you know they will never be written to later.

Reply:
Also, what do you think of calling `SmallestMissing` `PreviousSmallestMissing`, since it's for the gap before the current range? Is there a better word than 'previous'?

Reply:
The smallestmissing can be changed later so we cant do const for that